src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -107,7 +109,28 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 	) {
 		super();
 
+		// Initialize shell type, including a promise that completions can await for that resolves:
+		// - immediately if shell type
+		// - after a short delay if shell type gets set
+		// - after a long delay if it doesn't get set
 		this.shellType = shellType;
+		if (this.shellType) {
+			this._shellTypeInit = Promise.resolve();
+		} else {
+			const intervalTimer = this._register(new IntervalTimer());
+			const timeoutTimer = this._register(new TimeoutTimer());
+			this._shellTypeInit = new Promise<void>(r => {
+				intervalTimer.cancelAndSet(() => {
```

Comment:
could we/should we show the suggest widget with a loading indicator for this case? 

also, will this mean that I could request completions, see nothing, after let's say 3 seconds, the suggest widget pops up?

Reply:
Yes, but typically it shouldn't take near 5 seconds to init the shellType. I actually meant to do a follow up on this to set shell type immediately on terminal creation guessing via the executable. I thought I added that but need to review. This would just be the fallback mechanism.